api/v3/CivisocialUser.php

```diff
@@ -57,3 +67,65 @@ function civicrm_api3_civisocial_user_getFacebookEventInfo($params) {
   }
   return civicrm_api3_create_error("Not connected to Facebook.");
 }
+
+/**
+ * Makes a post to Facebook and/or Twitter
+ *
+ * @param array $params
+ *
+ * @return array
+ */
+function civicrm_api3_civisocial_user_updateStatus($params) {
+  civicrm_api3_verify_mandatory($params, NULL, array('post_content'));
+
+  $session = CRM_Core_Session::singleton();
+  $response = array();
+
+  if (isset($params['facebook'])) {
+    $pageId = $session->get('facebook_page_id');
+    $fbAccessToken = $session->get('facebook_page_access_token');
+    if ($pageId && $fbAccessToken) {
+      // Connected to page
+      $facebook = new CRM_Civisocial_OAuthProvider_Facebook($fbAccessToken);
+      // Check if token is still valid
+      $pageInfo = $facebook->get("{$pageId}?fields=name,picture");
+      if ($pageInfo) {
+        // Token valid
+        $post['message'] = $params['post_content'];
+        $result = $facebook->post("{$pageId}/feed", $post);
+        $response['facebook']['post_id'] = $result['id'];
+      }
+      else {
+        return civicrm_api3_create_error("Invalid Facebook access token.");
+      }
+    }
+    else {
+      return civicrm_api3_create_error("Not connected to Facebook.");
+    }
+  }
+
+  if (isset($params['twitter'])) {
+    $twitterId = $session->get('twitter_id');
+    $twitterAccessToken = $session->get('twitter_access_token');
+    if ($twitterId && $twitterAccessToken) {
+      // Connected to Twitter
+      $twitter = new CRM_Civisocial_OAuthProvider_Twitter($twitterAccessToken);
+      // Check if token is still valid
+      if ($twitter->isAuthorized()) {
+        $post['status'] = $params['post_content'];
+        $result = $twitter->post('statuses/update', $post);
+        if ($result && $result['id']) {
+          $response['twitter']['tweet_id'] = $result['id'];
+        }
+      }
+      else {
+        return civicrm_api3_create_error("Invalid Twitter access token.");
+      }
+    }
+    else {
+      return civicrm_api3_create_error("Not connected to Twitter.");
+    }
+  }
+
+  return $response;
+}
```

Comment:
looks like both set of code at line 84 and 107 are duplicate only variable names are changed could you combine them together?